iodata/test/test_factorial2.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+import numpy as np
+from .. import overlap
+
+factorial2 = overlap.factorial2
+
+
+class TestFactorial2(unittest.TestCase):
+    def test_integer_arguments(self):
+        self.assertEqual(factorial2(0, exact=True), 1)
+        self.assertEqual(factorial2(1, exact=True), 1)
+        self.assertEqual(factorial2(2, exact=True), 2)
+        self.assertEqual(factorial2(3, exact=True), 3)
+        self.assertEqual(factorial2(-1, exact=True), 1.0)
```

Comment:
Should the return value be an integer here?

Reply:
We have followed all your comments and changed the unittest to pytest_approx. We apologize for the mistake we were basing our work on your previous comment on the issue #308, which led us to believe  that we should use unittest. The test_factorial2 has been updated and posted in this pull request. If you don't see it, please let us know, and we will create a new pull request.

Thank you, and please let us know if you merge it.